src/file_hosting/s3_host.rs

```diff
@@ -0,0 +1,84 @@
+use crate::file_hosting::{DeleteFileData, FileHost, FileHostingError, UploadFileData};
+use async_trait::async_trait;
+use s3::bucket::Bucket;
+
+pub struct S3Host {
+    pub bucket: Bucket,
+}
```

Comment:
This should have a `new` function that takes a `Bucket` instead of having a `pub` field and using the struct constructor.  Using a constructor allows us to add new fields to this without breaking everywhere we create it.